rust/ql/test/library-tests/variables/variables.rs

```diff
@@ -26,9 +26,9 @@ fn mutable_variable() {
 
 fn mutable_variable_immutable_borrow() {
     let mut x = 1;
-    print_i64_ref(&x); // $ access=x
+    print_i64_ref(&x); // $ read_access=x
     x = 2; // $ write_access=x
-    print_i64_ref(&x); // $ access=x
+    print_i64_ref(&x); // $ read_access=x
```

Comment:
Hmm, if `print_i64_ref` took a mutable reference and wrote to it, would we still label it a `read_access`?

Reply:
Yes. It's sort of inaccurate. But I think that in order for read steps from `foo` to `&foo` to work the SSA library needs to treat `&foo` as a read. At least, from what I can see, it seems like the simplest and most straightforward way to handle `&` and `*`.

Reply:
OK.  I'd like to hear @hvitved 's opinion on this point as well, I'm not really sure what other languages do for this and why.

Reply:
See my earlier comment.